### PR TITLE
🍰 2019/kw22/alpha_data_import_status_schema_split

### PR DESCRIPTION
**Pull request by Tirokk**
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-29T13:43:12Z" title="Wednesday, May 29th 2019, 3:43:12 pm +02:00">May 29, 2019</time>_
_Merged <time datetime="2019-05-31T14:59:44Z" title="Friday, May 31st 2019, 4:59:44 pm +02:00">May 31, 2019</time>_
---

## 🍰 Pullrequest

- fixed several errors handling import
- split graphql schema into parts
- which data is imported which is not - a list

### Issues

- relates https://github.com/Human-Connection/Human-Connection/issues/692

### Todo

- [x] catalogue all the data not yet started to be imported
- [ ] import the data (will not be done in this PR)
- [ ] refactor graphql schema (will not be done in this PR)

### CODE REVIEW

File: backend/src/types/index.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      results = results.concat(findGqlFiles(file))
+    } else {
+      if (path.extname(file) === '.gql') {
+        // Is a gql file
+        results.push(file)
+      }
+    }
+  })
+  return results
+}
```
Comment (Tirokk) [inline backend/src/types/index.js:22]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-31T11:50:25Z" title="Friday, May 31st 2019, 1:50:25 pm +02:00">May 31, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,30 @@
+import fs from 'fs'
+import path from 'path'
+import { mergeTypes } from 'merge-graphql-schemas'
+
+const findGqlFiles = dir => {
+  var results = []
+  var list = fs.readdirSync(dir)
+  list.forEach(file => {
+    file = path.join(dir, file).toString('utf-8')
+    var stat = fs.statSync(file)
+    if (stat && stat.isDirectory()) {
+      // Recurse into a subdirectory
+      results = results.concat(findGqlFiles(file))
+    } else {
+      if (file.split('.').pop() === 'gql') {
+        // Is a gql file
+        results.push(file)
+      }
+    }
+  })
+  return results
+}
```


@ulfgebhardt great! Can we do sth. similar [here](https://github.com/Human-Connection/Human-Connection/blob/master/backend/src/graphql-schema.js#L41) ?

File: backend/src/types/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  var results = []
+  var list = fs.readdirSync(dir)
+  list.forEach(file => {
+    file = path.join(dir, file).toString('utf-8')
+    var stat = fs.statSync(file)
+    if (stat && stat.isDirectory()) {
+      // Recurse into a subdirectory
+      results = results.concat(findGqlFiles(file))
+    } else {
+      if (path.extname(file) === '.gql') {
```
Comment (Tirokk) [inline backend/src/types/index.js:15]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-31T13:05:23Z" title="Friday, May 31st 2019, 3:05:23 pm +02:00">May 31, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,30 @@
+import fs from 'fs'
+import path from 'path'
+import { mergeTypes } from 'merge-graphql-schemas'
+
+const findGqlFiles = dir => {
+  var results = []
+  var list = fs.readdirSync(dir)
+  list.forEach(file => {
+    file = path.join(dir, file).toString('utf-8')
+    var stat = fs.statSync(file)
+    if (stat && stat.isDirectory()) {
+      // Recurse into a subdirectory
+      results = results.concat(findGqlFiles(file))
+    } else {
+      if (file.split('.').pop() === 'gql') {
```


I guess there is a cleaner way to check for a file extension.

**Comment by Tirokk** [comment]
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-31T13:41:37Z" title="Friday, May 31st 2019, 3:41:37 pm +02:00">May 31, 2019</time>_

---


@roschaefer the given error makes no sense
Regarding to the neo4j [docu](https://neo4j.com/developer/kb/how-do-i-authenticate-with-cypher-shell-without-specifying-the-username-and-password-on-the-command-line/) the neo4j username and password are automatically set from environment variables called `NEO4J_USERNAME` and `NEO4J_PASSWORD` - the command uses those and the `.env` file does not override it by default.

Therefore they are either missing in your call or the database does not require authentication in that container?!

File: backend/src/types/index.js (hunk 1 above)
Comment (Tirokk) [reply backend/src/types/index.js:22]: > [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-05-31T14:06:21Z" title="Friday, May 31st 2019, 4:06:21 pm +02:00">May 31, 2019</time>_

---


yes

**Comment by Tirokk** [comment]
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-31T14:25:17Z" title="Friday, May 31st 2019, 4:25:17 pm +02:00">May 31, 2019</time>_

---


@ulfgebhardt I don't know why, but I added unrelated commits by accident, let me rebase that..

**Comment by Tirokk** [comment]
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-31T14:25:17Z" title="Friday, May 31st 2019, 4:25:17 pm +02:00">May 31, 2019</time>_

---


@ulfgebhardt I don't know why, but I added unrelated commits by accident, let me rebase that..

**Comment by Tirokk** [comment]
> [codecov[bot]](https://github.com/apps/codecov) **Authored by [codecov[bot]](https://github.com/apps/codecov)**
_<time datetime="2019-05-31T14:59:35Z" title="Friday, May 31st 2019, 4:59:35 pm +02:00">May 31, 2019</time>_

---


# [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/703?src=pr&el=h1) Report
> Merging [#703](https://codecov.io/gh/Human-Connection/Human-Connection/pull/703?src=pr&el=desc) into [master](https://codecov.io/gh/Human-Connection/Human-Connection/commit/284ca34e03fc707331c8fbe6891d1d19d5604d40?src=pr&el=desc) will **decrease** coverage by `0.12%`.
> The diff coverage is `0%`.

[Impacted file tree graph](https://codecov.io/gh/Human-Connection/Human-Connection/pull/703?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #703      +/-   ##
==========================================
- Coverage   21.27%   21.15%   -0.12%     
==========================================
  Files         161      162       +1     
  Lines        2261     2274      +13     
  Branches      113      113              
==========================================
  Hits          481      481              
- Misses       1752     1765      +13     
  Partials       28       28
```


| [Impacted Files](https://codecov.io/gh/Human-Connection/Human-Connection/pull/703?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [backend/src/seed/factories/badges.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/703/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvc2VlZC9mYWN0b3JpZXMvYmFkZ2VzLmpz) | `100% <ø> (ø)` | :arrow_up: |
| [backend/src/seed/factories/posts.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/703/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvc2VlZC9mYWN0b3JpZXMvcG9zdHMuanM=) | `100% <ø> (ø)` | :arrow_up: |
| [backend/src/seed/factories/users.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/703/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvc2VlZC9mYWN0b3JpZXMvdXNlcnMuanM=) | `100% <ø> (ø)` | :arrow_up: |
| [backend/src/resolvers/index.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/703/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvcmVzb2x2ZXJzL2luZGV4Lmpz) | `0% <0%> (ø)` | |
| [backend/src/types/index.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/703/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvdHlwZXMvaW5kZXguanM=) | `0% <0%> (ø)` | |

**Review by Mogge — APPROVED** [review]
Great job! @ulfgebhardt 

well done

**Pull request closed without merging**